Draft 1: Kubernetes documentation (#3115)
mattermost/docs: source/install/install-kubernetes-cluster.rst

.. _install-kubernetes-cluster:

Setting Up a Kubernetes Cluster
============================

Prerequisites
--------------

If you don’t have a production-ready Kubernetes cluster, version 1.12 or higher, you will need to set one up. If you already have a
Kubernetes cluster you can skip to Confirm Resource Requirements. Note that the

> Kubernetes-specific CLI, `kubectl <https://kubernetes.io/docs/reference/kubectl/overview/>`__, also needs to be installed and configured.

Reviewer: ```suggestion
Kubernetes-specific CLI tool, `kubectl <https://kubernetes.io/docs/reference/kubectl/overview/>`__, also needs to be installed and configured.
```

> It’s recommended that you have an understanding of basic Kubernetes concepts (such as pods) and

Reviewer: ```suggestion
It’s recommended that you have an understanding of basic Kubernetes concepts (such as deployments, pods) and
```

> actions (such as start, stop, redeploy, update). It's also advisable to consult the

Reviewer: ```suggestion
actions (such as applying manifests, viewing pod logs). It's also advisable to consult the
```

`official Kubernetes setup documentation <https://kubernetes.io/docs/setup/>`__ on how to set up a cluster in your
environment. If you’re unsure about which environment you want to use for your Kubernetes cluster, we
suggest using a managed service such as as `Amazon EKS <https://aws.amazon.com/eks/>`__, `Azure Kubernetes Service <https://azure.microsoft.com/en-ca/services/kubernetes-service/>`__, `Google Kubernetes Engine <https://cloud.google.com/kubernetes-engine/>`__,
or `DigitalOcean Kubernetes <https://www.digitalocean.com/products/kubernetes/>`__.

Confirm Resource Requirements
-----------------------------

Running Mattermost in Kubernetes requires different resources based on your total number of users.

> The table below details the minimum Kubernetes cluster resources that Mattermost requires at different scales.

Reviewer: It's important that we mention these are just guidelines and not hard and fast resource requirements
